src/main/io/osd.c

```diff
@@ -1564,6 +1580,15 @@ static void osdShowStats(uint16_t endBatteryVoltage)
         osdDisplayStatisticLabel(top++, "MAX DISTANCE", buff);
     }
 
+    if (osdStatGetState(OSD_STAT_TOTAL_DISTANCE)) {
+        uint32_t distanceFlown = 0;
+#ifdef USE_GPS
```

Comment:
Regarding the entire block being conditional: I was following the same reasoning as with MAX DISTANCE. Even if you don't have a (working/enabled) GPS, you can enable the stat and it will show zero.
Regarding checking the feature: you're right but it actually would be even more correct to check if we have a GPS_FIX and GPS_FIX_HOME

Reply:
No, because that would lead to confusion. User's wouldn't understand why the stats are sometimes missing if they have GPS enabled.  Having them be 0 in this case is correct and is logical if the user doesn't have a GPS fix.

Reply:
@TonyBlit: Showing GPS statistics on firmware that cannot even support GPS (i.e. `USE_GPS`  is not defined) will be even more confusing. And as I said (and confirmed by @etracer65), the runtime check should be for `featureIsEnabled(FEATURE_GPS)` so that the stat is shown if the user has GPS enabled, even if they don't have a GPS fix when the stat is shown (think a user taking off / landing on a balcony, but flying outside).

Reply:
Understood. Do I leave MAX DISTANCE as it is? Or wrap it with USE_GPS as well?

Reply:
If you're already in there it would be helpful to fix it. Note however that your new stat needs to go at the bottom.  Again note the comment at the top of the function about keeping the ordering consistent with how the stats will be presented in the configurator.

Reply:
Done. MAX SPEED also needed some wrapping. I also updated the code to use featureIsEnabled so all stats use the same logic: they will all show zero if there's no GPS_FIX.
I've noticed that losing GPS connection is not actively handled, resulting in showing old data. I'll further think about it, and open a new PR if needed.